src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -351,34 +351,24 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_aggregate_node(
 
   const auto input_operator = translate_node(node->left_input());
 
-  // Create AggregateColumnDefinitions from AggregateExpressions
-  // All aggregate_pqp_expressions have to be AggregateExpressions and their argument() has to be a PQPColumnExpression
-  std::vector<AggregateColumnDefinition> aggregate_column_definitions;
-  aggregate_column_definitions.reserve(aggregate_node->aggregate_expressions_begin_idx);
+  std::vector<std::shared_ptr<AggregateExpression>> pqp_aggregate_expressions;
+  pqp_aggregate_expressions.reserve(aggregate_node->node_expressions.size() -
+                                    aggregate_node->aggregate_expressions_begin_idx);
   for (auto expression_idx = aggregate_node->aggregate_expressions_begin_idx;
        expression_idx < aggregate_node->node_expressions.size(); ++expression_idx) {
-    const auto& expression = aggregate_node->node_expressions[expression_idx];
+    const auto& lqp_expression = aggregate_node->node_expressions[expression_idx];
 
-    Assert(
-        expression->type == ExpressionType::Aggregate,
-        "Expression '" + expression->as_column_name() + "' used as AggregateExpression is not an AggregateExpression");
-
-    const auto& aggregate_expression = std::static_pointer_cast<AggregateExpression>(expression);
-
-    // Always resolve the aggregate to a column, even if it is a Value. The Aggregate operator only takes columns as
-    // arguments
-    const auto column_expression = dynamic_cast<const LQPColumnExpression*>(&*aggregate_expression->argument());
-    auto argument_column_id = ColumnID{};
-    if (column_expression && column_expression->column_reference.original_column_id() == INVALID_COLUMN_ID) {
-      // Handle COUNT(*)
-      argument_column_id = INVALID_COLUMN_ID;
-    } else {
-      argument_column_id = node->left_input()->get_column_id(*aggregate_expression->argument());
-    }
-    aggregate_column_definitions.emplace_back(argument_column_id, aggregate_expression->aggregate_function);
+    Assert(lqp_expression->type == ExpressionType::Aggregate,
```

Comment:
Shouldn't a `DebugAssert` suffice here?

Reply:
The LQPTranslator currently only uses Assert. As those are the asserts that come up regularly when I break things and as the translator cost is quite low, I'd keep it that way.